### ISSUE TITLE
Add actions from schema.org as GMail supports them

### ISSUE THREAD

**Issue by MeiKatz** [nextcloud/mail#5881]
### Feature Request

Are there any plans for supporting schema.org actions like [those that GMail is supports](https://developers.google.com/gmail/markup/overview)? Or is there anything that would block adding support for them?

If not, I would like to investigate it this topic and maybe add the support. Maybe not for all kinds of actions at once, but stepwise – first only the easy ones (the basic ones) and maybe the more complex later.

There might also be an option to accept or deny event requests, and adding the related event to the personal calendar at NextCloud if accepted (or do we have this already? I am not sure about this).

#### Extract from the linked document
> Gmail supports the following types of actions and interactive cards:
> - One-click Action for just about anything that can be performed with a single click
> - Go-to Action for more complex interactions
> - Flight interactive cards
>
> Gmail uses the same schema.org markup to highlight the most important information within an email, and allow user to act on it.
> - Bus Highlight
> - Car Rental Highlight
> - Flight Highlight
> - Event Highlight for events with tickets
> - Order and parcel delivery Highlights for shopping
> - Hotel Highlight for hotel reservations
> - Invoice Highlight
> - Restaurant Highlight for restaurant reservations
> - Train Highlight

**Comment by miaulalala** [comment]
IIRC here is something to extract events from emails already, but I'm not sure if this is based on a Schema or not.

I like this idea. Not sure on what to do with the extracted info. Displaying it as a highlight as GMail does would be nice.

**Comment by MeiKatz** [comment]
> IIRC here is something to extract events from emails already, but I'm not sure if this is based on a Schema or not.
> 
> I like this idea. Not sure on what to do with the extracted info. Displaying it as a highlight as GMail does would be nice.

Yes, I think the first step would be to show the information in an appropriate way. Maybe we find additional use cases for that later on.

**Comment by vkrause** [comment]
There's already something shown for flights, trains and events. The extractor also provides data for bus, car rental, hotel and restaurant reservations, so those mainly lack the UI part. Order and invoice data is currently discarded in the extractor as it's mainly focused on travel use-cases, but that can be changed if needed.

The data is provided in the schema.org JSON format (with one extension for specifying IANA timezones where those could be determined), and is taken from schema.org annotations in emails but also other structured sources such as ticket barcodes or attached Apple Wallet passes, as well as a number of vendor-specific extractor scripts.

**Comment by MeiKatz** [comment]
> There's already something shown for flights, trains and events. The extractor also provides data for bus, car rental, hotel and restaurant reservations, so those mainly lack the UI part. Order and invoice data is currently discarded in the extractor as it's mainly focused on travel use-cases, but that can be changed if needed.
> 
> The data is provided in the schema.org JSON format (with one extension for specifying IANA timezones where those could be determined), and is taken from schema.org annotations in emails but also other structured sources such as ticket barcodes or attached Apple Wallet passes, as well as a number of vendor-specific extractor scripts.

Okay, where can I find the code for this? I only know about the stuff in `src/components/itineraryEventReservation.vue` ...

**Comment by marxistvegan** [comment]
Any chance to bring life into this again? it would be a great feature for us trying to get off of Google Suite and Microsoft

**Comment by ChristophWurst** [comment]
@marxistvegan sure! Would you be interested in coding this feature?

**Comment by hhappel** [comment]
Just stumbled upon this.

We've been working an major changes / extensions of the existing Schema.org support for Nextcloud Mail (see https://github.com/audriga/nextcloud-mail/pull/1 / https://github.com/nextcloud/mail/pull/9291).

While afair actions mentioned here are not yet part of it, it might make sense to join forces here (resp. we'll try to include it into a further iteration of our NC Mail changes).

Some further cross-links:
- https://github.com/audriga/roundcube-structured-email
- https://github.com/mjl-/mox/issues/143
- https://github.com/cypht-org/cypht/issues/696
- https://github.com/the-djmaze/snappymail/issues/1422
